components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsConfiguration.java

```diff
@@ -364,4 +380,44 @@ public Integer getProxyPort() {
     public void setProxyPort(Integer proxyPort) {
         this.proxyPort = proxyPort;
     }
+
+    /**
+     * Since *Camel 2.20*. Only for FIFO queues. Strategy for setting the messageGroupId on the message.
+     * Can be one of the following options: *useConstant*, *useExchangeId*, *useHeaderValue*.
+     * For the *useHeaderValue* option, the value of header "CamelAwsMessageGroupId" will be used.
```

Comment:
I'd have expected you to use the exchange properties rather than the message headers as the headers tend to be copied verbatim to endpoints where as exchange properties stay within the JVM unless explicitly read, which you are doing in the `configureFifoAttributes` method via `HeaderValueStrategy`.

Reply:
Like this?

For the *useHeaderValue* option, the value of header SqsConstants.MESSAGE_GROUP_ID_HEADER will be used.

Reply:
Sorry, just seen this response; something like that, yes. You can also use an `@see` javadoc annotation, which will generate a href to the constant when the javadoc tool is used.